File: Data Structures & Algorithms/top-k-elements-in-list/submission-7.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        
        out = []

        hashmap = {}

        for i in nums: 
            if i not in hashmap: 
                hashmap[i] = 1
            elif i in hashmap:
                hashmap[i] += 1
        
        vallst = sorted(hashmap.values())[::-1][:k]

        for key in hashmap: 
            if hashmap[key] in vallst:
                out.append(key)
                
        return out
        
        

File: Data Structures & Algorithms/two-integer-sum/submission-7.java
class Solution {
    /*
        we can store the the value as a key and the index as the value in a hashmap
        every iteration we can check if the difference between the target and the current value that
        the pointer is on is in the hashmap, if it is you return the 2 indexs, otherwise you would
       just keep adding to the dict
    */

    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();

        for (int idx = 0; idx < nums.length; idx++) {
            int currIdxVal = nums[idx];
            int difference = target - currIdxVal;

            if (map.containsKey(difference)) {
                // we want to return the indexes of these 2
                return new int[] {map.get(difference), idx};
            }
            // continue putting elements in the map
            map.put(currIdxVal, idx);
        }
        return new int[] {};
    }
}
